docs/future/compatibility.md

```diff
@@ -0,0 +1,174 @@
+understand this format as well. Standardization is a good thing, and could allow tools to be built that maybe
+don't even exist yet.
+
+
+## Proposal
+
+
+### React Standard Tree (RST)
+
+This proposal hinges on a standard tree specification. Keep in mind that this tree needs to account for more
+than what is currently satisfied by the output of something like `react-test-renderer`, which is currently
+only outputting the "host" nodes (ie, HTML elements). We need a tree format that allows for expressing a full
+react component tree, including composite components.
+
+```js
+// Strings and Numbers are rendered as literals.
+type LiteralValue = string | number
+
+// A "node" in an RST is either a LiteralValue, or an RSTNode
+type Node = LiteralValue | RSTNode
+
+// An RSTNode has this specific shape
+type RSTNode = {|
+  // Either a string or a function. A string is considered a "host" node, and
+  // a function would be a composite component. It would be the component constructor or
+  // an SFC in the case of a function.
+  type: string | function
+  
+  // The props object passed to the node, which will include `children` in its raw form,
+  // exactly as it was passed to the component.
+  props: object
+  
+  // The backing instance to the node. Can be null in the case of "host" nodes and SFCs.
+  // React alternative libs can choose to provide whatever concept here applies.  If it is
+  // provided, Enzyme will expect it to follow the ComponentInterface type specified below
+  instance: ComponentInstance?
+  
+  // This is a transformed version of `props.children`. It is an array or null. If it's an array,
+  // it's an array of Nodes, meaning that they are LiteralNodes or RSTNodes. This means that 
+  // non-rendering values such as `false`, `null`, and `[]` wouldn't make it in here. This is also
+  // a "flat" array.
+  children: [Node]?
+  
+  // This can feel similar to children, but it's not the same thing at all. For a given node,
+  // this corresponds to the result of the `render` function with the provided props, but transformed
+  // into an RST.  For "host" nodes, this will always be `null`.
+  rendered: Node?
+|}
+```
+
+
+A `ComponentInstance` is going to be expected by enzyme to follow a certain interface in order
+for certain API methods to actually work, but they don't *need* to. Those methods just won't be
+guaranteed to work in those cases.
+
+
+So far, that interface is:
+```js
+type ComponentInstance = {
+  state: object?
+  context: object?
+  setState: function
+}
+```
+
+
+### Enzyme Adapter Protocol
+
+**Definitions:**
+
+An `Element` is considered to be whatever data structure is returned by the JSX pragma being used. In the 
+react case, this would be the data structure returned from `React.createElement`
+
+
+```js
+type RendererOptions = {
+  // I'm not 100% sure about this, but I think with an API like this, we could implement
+  // `mount` and `shallow` with the same renderer.
+  ?isLeaf(Element): bool
```

Comment:
I always imagined shallow rendering as effectively treating composite components as host components, so it would make sense for it to "render `props.children`" I think